cmd/search_log.go

```diff
@@ -354,10 +364,91 @@ func latencyAggregateFn(cmd *cobra.Command, args []string) {
 			printCsv([][]string{sortedRecords[i]})
 			worstQueriesTrs = fmt.Sprintf("%s%s", worstQueriesTrs, printHtmlTr(sortedRecords[i], false, ""))
 		}
-		worstQueriesHtmlPart = fmt.Sprintf("<h3>%d worst queries</h3><table>%s</table>", worstQueriesPrintCnt, worstQueriesTrs)		
+		worstQueriesHtmlPart = fmt.Sprintf("<h3>%d worst queries</h3><table>%s</table>", worstQueriesPrintCnt, worstQueriesTrs)
 	}
 	finalHtml := fmt.Sprintf("%s%s", operationsHtmlPart, worstQueriesHtmlPart)
 	err = ioutil.WriteFile(latencyOutputHtml, []byte(finalHtml), 0644)
 	utils.Must(err)
 	log.Info("HTML printed.")
 }
+
+func queriesAggregateFn(cmd *cobra.Command, args []string) {
+	logger := initLogger()
+	printCsv([][]string{{"Term", "Count", "SortType", "Language"}})
+	SLICES := 100
+	RESULTS_FOR_LANGUAGE := 1000
+	gteStr := null.NewString("", false)
+	languages := []string{consts.LANG_HEBREW, consts.LANG_ENGLISH, consts.LANG_RUSSIAN} // Currently only the 3 main languages are supported due to language recognition issues
+	queryCountByLang := map[string]map[string]int{}                                     // Language -> [Search Term -> Count]
+	if lastDays > 0 {
+		gteStr = null.StringFrom(fmt.Sprintf("now-%dd/d", lastDays))
+	}
+	for i := 0; i < SLICES; i++ {
+		s := elastic.NewSliceQuery().Id(i).Max(SLICES)
+		queries, err := logger.GetLattestQueries(s, gteStr, null.NewBool(false, false))
```

Comment:
What GetLattestQueries does?

Reply:
How did you test it works well?

Reply:
GetLattestQueries bring all search queries until the given date (gteStr param).

Reply:
Tested it locally with small amount of data